### PR TITLE
Update docs relating to translation extraction

### LINKED ISSUE #15061: [FrameworkBundle][Translation] Extract translation IDs from all of src
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#40229
| PR author(s) | @natewiebe13
| Merged in    | 5.3-dev

### PR DESCRIPTION
**Pull request by natewiebe13**
Update translation extraction docs based on changes from https://github.com/symfony/symfony/pull/40229

Fixes #15061

### CODE REVIEW

File: translation.rst [hunk 1]
```diff
@@ -386,6 +388,9 @@ The ``translation:update`` command looks for missing translations in:
   :ref:`twig.paths <config-twig-paths>` config options);
 * Any PHP file/class that injects or :doc:`autowires </service_container/autowiring>`
   the ``translator`` service and makes calls to the ``trans()`` method.
+* Any PHP file/class stored in the ``src/`` directory that creates a
+  translatable object using the constructor or the ``t()`` method or calls
```
Comment (Nyholm) [inline translation.rst]: Did you forget to link to the section about "translatable object"?

**Review by Nyholm — CHANGES_REQUESTED** [review]
Thank you. 

I like this. Please link up to the "Translatable object" section, that will make the sentence read better.

File: translation.rst (hunk 1 above)
Comment (natewiebe13) [reply translation.rst]: Link added

File: translation/debug.rst [hunk 2]
```diff
@@ -19,9 +19,10 @@ command helps you to find these missing or unused translation messages templates
 
 .. caution::
 
-    The extractors can't find messages translated outside templates, like form
-    labels or controllers. Dynamic translations using variables or expressions
-    in templates are not detected either:
+    The extractors can't find messages translated outside templates (like form
+    labels or controllers) unless using :ref:`translatable-objects` or calling
```
Comment (Nyholm) [inline translation/debug.rst:23]: Will this find a reference in a different file?

**Review by Nyholm — APPROVED** [review]
Im happy with this PR.  

I let the docs team help out with the syntax.

File: translation/debug.rst (hunk 2 above)
Comment (natewiebe13) [reply translation/debug.rst:23]: It'll be good for the docs team to confirm, but this reference does work if I build the docs locally with Docker, so I think it should work.

Comment (wouterj) [reply translation/debug.rst:23]: Yes, this works :)

**Comment by wouterj** [comment]
Thank you @natewiebe13 for providing the docs for your code contribution! Fyi, I've added small notes that this feature is new in 5.3 in dc9e61344

**Pull request merged**